src/export/render.rs

```diff
@@ -312,9 +312,42 @@ fn render_shape(
         canvas.fill_path(&path, &paint, rule, ts, mask);
     }
 
-    if let Some(Stroke { paint, thickness }) = &shape.stroke {
+    if let Some(Stroke { paint, thickness, line_cap, line_join, dash_pattern, miter_limit }) = &shape.stroke {
+        let line_cap = match line_cap {
+            LineCap::Butt => sk::LineCap::Butt,
+            LineCap::Round => sk::LineCap::Round,
+            LineCap::Square => sk::LineCap::Square,
+        };
+        let line_join = match line_join {
+            LineJoin::Miter => sk::LineJoin::Miter,
```

Comment:
maybe we want some From impl's for typst -> tiny-skia and typst -> pdf-writer types?